### PR TITLE
feat: improve client performance

### PR DESCRIPTION
**Pull request by Tomvbe**

### CODE REVIEW

File: ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/inmemory/InMemoryTreeNodeRecordRepository.java [hunk 1]
```diff
@@ -21,16 +22,22 @@ public void saveTreeNodeRecord(TreeNodeRecord treeNodeRecord) {
...
 				immutable.add(treeNodeRecord);
-				notVisited.remove(treeNodeRecord);
+				almostImmutable.remove(treeNodeRecord);
 			}
-		}
+            case IMMUTABLE_WITH_UNPROCESSED_MEMBERS -> {
+				almostImmutable.add(treeNodeRecord);
+				notVisited.remove(treeNodeRecord);
+            }
+        }
```
Comment (Yalz) [inline ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/inmemory/InMemoryTreeNodeRecordRepository.java:33]: Spacing issue or just in GH ?

File: ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/inmemory/InMemoryTreeNodeRecordRepository.java [hunk 2]
```diff
@@ -54,12 +67,15 @@ public void destroyState() {
...
+	public Optional<TreeNodeRecord> getTreeNodeRecordWithStatusAndEarliestNextVisit(TreeNodeStatus treeNodeStatus) {
 		return switch (treeNodeStatus) {
 			case NOT_VISITED -> notVisited.isEmpty() ? Optional.empty() : Optional.of(notVisited.get(0));
 			case MUTABLE_AND_ACTIVE ->
 				mutableAndActive.isEmpty() ? Optional.empty() : Optional.of(mutableAndActive.poll());
-			case IMMUTABLE -> immutable.isEmpty() ? Optional.empty() : immutable.stream().findFirst();
+			case IMMUTABLE_WITH_UNPROCESSED_MEMBERS
+					-> almostImmutable.isEmpty() ? Optional.empty() : almostImmutable.stream().findFirst();
+			case IMMUTABLE_WITHOUT_UNPROCESSED_MEMBERS
+					-> immutable.isEmpty() ? Optional.empty() : immutable.stream().findFirst();
```
Comment (Yalz) [inline ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/inmemory/InMemoryTreeNodeRecordRepository.java:78]: Previous code = first "->" , then newline. New code has new layout?

**Review by Yalz — APPROVED** [review]

File: ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/inmemory/InMemoryTreeNodeRecordRepository.java (hunk 1 above)
Comment (Tomvbe) [reply ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/inmemory/InMemoryTreeNodeRecordRepository.java:33]: gh

File: ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/inmemory/InMemoryTreeNodeRecordRepository.java (hunk 2 above)
Comment (Tomvbe) [reply ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/repository/inmemory/InMemoryTreeNodeRecordRepository.java:78]: new code was too long to be readable on one line without a widescreen

**Pull request merged**